### PR TITLE
NO-JIRA: Added doc/how-to-add-new-output to help developers.

### PR DESCRIPTION
**Pull request by alanconway**
/cc @vimalk78

### CODE REVIEW

File: docs/how-to-add-new-output.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+The ClusterLogForwarder API is defined by a collection of Go struct types.
+
+[type ClusterLogForwarderSpec struct][cluster_log_forwarder] specifies
+* inputs to select logs
+* outputs to forward logs to remote targets
+* pipelines to connect inputs to outputs
+
+It supports an open-ended set of "output types", such as Loki, Kafka, Elasticsearch and so on.
+
+[type ClusterLogForwarderOutputSpec struct][cluster_log_forwarder] specifies common Name, Type, URL, Secret fields used by all outputs. The Secret fields references a K8s Secret object used to store any sensitive authentication or authorization data needed by the output.
```
Comment (vimalk78) [inline docs/how-to-add-new-output.md:14]: the name of the type is `OutputSpec`.
should we specify using go package ? `github.com/openshift/cluster-logging-operator/apis/logging/v1/OutputSpec`

File: docs/how-to-add-new-output.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
+# How to add a new output type
+
+## Understanding the API
+
+The ClusterLogForwarder API is defined by a collection of Go struct types.
+
+[type ClusterLogForwarderSpec struct][cluster_log_forwarder] specifies
+* inputs to select logs
```
Comment (vimalk78) [inline docs/how-to-add-new-output.md:8]: inputs to select logs using `InputSpec`

File: docs/how-to-add-new-output.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+* outputs to forward logs to remote targets
```
Comment (vimalk78) [inline docs/how-to-add-new-output.md:9]: outputs to forward logs to remote targets using `OutputSpec` and `OutputTypeSpec`

File: docs/how-to-add-new-output.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+* pipelines to connect inputs to outputs
+
+It supports an open-ended set of "output types", such as Loki, Kafka, Elasticsearch and so on.
+
+[type ClusterLogForwarderOutputSpec struct][cluster_log_forwarder] specifies common Name, Type, URL, Secret fields used by all outputs. The Secret fields references a K8s Secret object used to store any sensitive authentication or authorization data needed by the output.
+
+[type OutputTypeSpec struct][output_types] MAY add extra fields that are unique to the new output type.
```
Comment (vimalk78) [inline docs/how-to-add-new-output.md:16]: in addition we can say :
define a golang struct for the new type, and add it to the `OutputTypeSpec`

**Review by vimalk78 — COMMENTED** [review]
added some comments (nits)
In the top section we can say that "adding a output involves two steps 1) adding the type to the CLF spec , and 2) generating configuration for the supported collectors."
then have a separate section for each, with its own details.

**Comment by alanconway** [comment]
@vimalk78  thanks, good suggestions - applied them all. LGTM?

**Review by jcantrill — COMMENTED** [review]
/lgtm

**Pull request merged**